sqlobject/src/main/java/org/jdbi/v3/sqlobject/DefaultGeneratedKeyMapper.java

```diff
@@ -32,11 +32,16 @@
 
     @Override
     public Object map(ResultSet r, StatementContext ctx) throws SQLException {
-        ColumnMapper<?> columnMapper = ctx.findColumnMapperFor(returnType)
-                .orElseThrow(() -> new IllegalStateException("No column mapper for " + returnType));
-
-        return "".equals(columnName)
+        ColumnMapper<?> columnMapper = ctx.findColumnMapperFor(returnType).orElse(null);
+        if (columnMapper != null) {
+            return "".equals(columnName)
                 ? columnMapper.map(r, 1, ctx)
                 : columnMapper.map(r, columnName, ctx);
+        }
+        RowMapper<?> rowMapper = ctx.findRowMapperFor(returnType).orElse(null);
```

Comment:
Unfortunately if we do that, we break the `columnName` annotation handling passed in on our constructor.